crates/ruff_linter/src/rules/flake8_builtins/rules/stdlib_module_shadowing.rs

```diff
@@ -99,3 +100,36 @@ pub(crate) fn stdlib_module_shadowing(
         TextRange::default(),
     ))
 }
+
+/// Return the longest prefix of `path` between `settings.src` and `settings.project_root`.
+fn get_prefix<'a>(settings: &'a LinterSettings, path: &Path) -> Option<&'a PathBuf> {
+    let mut prefix = None;
+    let mut prefix_len = 0;
+    for dir in settings.src.iter().chain([&settings.project_root]) {
+        let len = dir.as_os_str().len();
+        if len > prefix_len && path.starts_with(dir) {
+            prefix = Some(dir);
+            prefix_len = len;
+        }
```

Comment:
I think we should use natural ordering here rather than just comparing by byte length. 
```suggestion
        let len = dir.as_os_str().len();
        if path.starts_with(dir) && prefix.is_none_or(|existing| existing < dir) {
            prefix = Some(dir);
        }
```

Reply:
Oh good catch, and now I can get rid of the separate `len` tracking!

Reply:
Oh, I just got an MSRV warning that `is_none_or` is in 1.82 but our MSRV is 1.80. Is that right, or is something in my local setup out of date?